Lib/asyncio/base_events.py

```diff
@@ -107,9 +107,9 @@ def _ipaddr_info(host, port, family, type, proto, flowinfo=0, scopeid=0):
             host is None:
         return None
 
-    if type == socket.SOCK_STREAM:
+    if type & socket.SOCK_STREAM:
         proto = socket.IPPROTO_TCP
-    elif type == socket.SOCK_DGRAM:
+    elif type & socket.SOCK_DGRAM:
         proto = socket.IPPROTO_UDP
     else:
```

Comment:
Continuing to exclude SOCK_RAW here actually seems like the right thing to do (since raw sockets don't have IP addresses).

If omitting this part of the change doesn't fix the original bug report, then I'd suggest adding `SOCK_RAW` as its own branch saying something like:

```
elif type == socket.SOCK_RAW:
    # asyncio mishandles raw sockets if `_ipaddr_info` returns `None`,
    # so fib about the underlying protocol
    proto = socket.IPPROTO_UDP
```

Reply:
This change isn't required to fix our specific problem. I feel fewer side effects would come from preserving the current behaviour.